### DISCUSSION TITLE
Units have name & nick name

### DISCUSSION THREAD

**Discussion by bevans2000** [mars-sim/mars-sim#444] [Q&A] [answered]
Why do the Units have a getName & getNickName, it is confusing. Particularly for Building where the name attribute is the type of Building and the nick name is actually the locally unique name.  Name should be the unique name in all cases so the `Building` class is in error. 
Then the code could use getName which makes more sense.

**Comment by mokun** [comment, marked as answer]
Before and in v3.06, a building has no unique name. 

e.g. If there are two Inflatable Greenhouses, they both have the same name `Inflatable Greenhouse`. getName() is not useful at all.

Starting v3.07, we added buildingType and getBuildingType() to denote the type of building. 

So if there are two Inflatable Greenhouses, they both have the same building type `Inflatable Greenhouse`. getBuildingType() becomes useful for this distinction. 

At the same time, we added nickName and getNickName() to distinguish between `Inflatable Greenhouse 1` and `Inflatable Greenhouse 2`.

So name and getName() should be deprecated and may be we should get rid of it.

**Comment by bevans2000** [comment]
Yeap I get that, getBuildingType is the correct way to go. Therfore the getName in Building should return the unique name? I think yes. Then getNickName becomes a duplicate of getName and can go to tidy the code.

**Reply by mokun** [reply]
Agree 👍
